File: order-ms/src/main/java/com/nexeum/orderms/controller/OrderController.java
package com.nexeum.orderms.controller;

import com.nexeum.orderms.dto.request.CancelOrderRequest;
import com.nexeum.orderms.dto.request.CreateOrderRequest;
import com.nexeum.orderms.dto.request.GetAllOrderRequest;
import com.nexeum.orderms.service.OrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/order")
public class OrderController {
    @Autowired
    private OrderService orderService;

    @PostMapping("/get-all-orders")
    public ResponseEntity<?> getAllOrders(@RequestBody GetAllOrderRequest request) {
        ResponseEntity<?> response = orderService.getAllOrders(request);
        return response;
    }

    @PostMapping("/create-order")
    public ResponseEntity<?> createOrder(@RequestBody CreateOrderRequest request) {
        ResponseEntity<?> response = orderService.createOrder(request);
        return response;
    }

    @PostMapping("/cancel-order")
    public ResponseEntity<?> cancelOrder(@RequestBody CancelOrderRequest request) {
        ResponseEntity<?> response = orderService.cancelOrder(request);
        return response;
    }
}

File: order-ms/controller/OrderController.cs
using Microsoft.AspNetCore.Mvc;
using order_ms.dto.request;
using order_ms.service;

namespace order_ms.controller
{
    [ApiController]
    [Route("api/v1/order")]
    public class OrderController : ControllerBase
    {
        private readonly IOrderService _orderService;

        public OrderController(IOrderService orderService) => _orderService = orderService;

        [HttpGet("get_all_orders")]
        public async Task<IActionResult> GetAllOrders() => await _orderService.GetAllOrders();

        [HttpPost("create_order")]
        public async Task<IActionResult> CreateOrder([FromBody] CreateOrderRequest request) => await _orderService.CreateOrder(request);

        [HttpPost("cancel_order")]
        public async Task<IActionResult> CancelOrder([FromBody] CancelOrderRequest request) => await _orderService.CancelOrder(request);
    }
}